react-instantsearch-hooks/facet-dropdown/src/Dropdown.tsx

```diff
@@ -0,0 +1,178 @@
+import {
+  PropsWithChildren,
+  useCallback,
+  useEffect,
+  useRef,
+  useState,
+} from 'react';
+import type { SearchResults } from 'algoliasearch-helper';
+import type {
+  CurrentRefinementsConnectorParamsRefinement,
+  CurrentRefinementsRenderState,
+} from 'instantsearch.js/es/connectors/current-refinements/connectCurrentRefinements';
+import type { UiState } from 'instantsearch.js';
+import {
+  useCurrentRefinements,
+  useInstantSearch,
+} from 'react-instantsearch-hooks-web';
+
+import { Panel } from './Panel';
+import { useCloseDropdown } from './useCloseDropdown';
+import { useLockedBody } from './useLockedBody';
+import { useMediaQuery } from './useMediaQuery';
+import { capitalize, cx, getFirstChildPropValue } from './utils';
+
+import './Dropdown.css';
+
+export type DropdownProps = PropsWithChildren<{
+  buttonText?: string | ((options: DropdownButtonTextOptions) => string);
+  classNames?: Partial<DropdownClassNames>;
+  closeOnChange?: boolean | (() => boolean);
+}>;
+
+export type DropdownButtonTextOptions = {
+  results: SearchResults;
+  uiState: UiState;
+  refinements: CurrentRefinementsConnectorParamsRefinement[];
+};
+
+export type DropdownClassNames = {
+  button: string;
+  buttonRefined: string;
+  closeButton: string;
+  mobileTitle: string;
+};
+
+type MiddlewareProps = Pick<DropdownProps, 'closeOnChange'> & {
+  isOpened: boolean;
+  close: () => void;
+};
+
+function getAttributeRefinements(
+  attribute: string,
+  items: CurrentRefinementsRenderState['items']
+) {
+  const item = items.find((item) => item.attribute === attribute);
+  return item?.refinements || [];
+}
+
+function DropdownMiddleware({
+  isOpened,
+  closeOnChange,
+  close,
+}: MiddlewareProps) {
+  const { use } = useInstantSearch();
```

Comment:
Since the `useInstantSearch` hook now relies on the "render" event from InstantSearch.js (https://github.com/algolia/react-instantsearch/pull/3554), an error is triggered because I use the hook in multiple components (in the Dropdown component, Dropdown middleware, and Panel component).

Not sure how to fix this?

![image](https://user-images.githubusercontent.com/662153/182193090-ed2dea93-e35d-4b18-bea6-b3ab7a349a77.png)

Reply:
Thanks for logging it!